examples/basic/adapter.ts

```diff
@@ -0,0 +1,45 @@
+import { DiscordGatewayAdapterCreator, DiscordGatewayAdapterLibraryMethods } from '../../';
+import { VoiceChannel, Snowflake, Client, Constants } from 'discord.js';
+import {
+	GatewayVoiceServerUpdateDispatchData,
+	GatewayVoiceStateUpdateDispatchData,
+} from 'discord-api-types/v8/gateway';
+
+const adapters = new Map<Snowflake, DiscordGatewayAdapterLibraryMethods>();
+const trackedClients = new Set<Client>();
+
+/**
+ * Tracks a Discord.js client, listening to VOICE_SERVER_UPDATE and VOICE_STATE_UPDATE events.
+ * @param client The Discord.js Client to track
+ */
+function trackClient(client: Client) {
+	if (trackedClients.has(client)) return;
+	trackedClients.add(client);
+	client.ws.on(Constants.WSEvents.VOICE_SERVER_UPDATE, (payload: GatewayVoiceServerUpdateDispatchData) => {
```

Comment:
I used the ones exported by Discord.js here since I'm listening to events from Discord.js's `ws` (if the code wasn't directly related to Discord.js, then I would have used `discord-api-types`)

Reply:
In a way, it kind of is.

`client.ws`'s events are emitted with the event name being `event.t`, and data being `event.d`, so yes, they directly map to `discord-api-types`'s values.